sklearn/tests/test_pipeline.py

```diff
@@ -1647,3 +1647,35 @@ def test_feature_union_getitem_error(key):
     msg = "Only string keys are supported"
     with pytest.raises(KeyError, match=msg):
         union[key]
+
+
+def test_feature_union_feature_names_in_():
+    """Ensure feature union has `.feature_names_in_` attribute if `X` has a
+    `columns` attribute.
+
+    Test for #24754"""
+    pytest.importorskip("pandas")
+
+    X, _ = load_iris(as_frame=True, return_X_y=True)
+    X_train, X_test = train_test_split(X, random_state=0)
```

Comment:
Since we are only testing the API for `feature_names_in_` and `X_test` is not used, we can use `X` directly in this test without `train_test_split`.

Reply:
See commit [073daf7](https://github.com/scikit-learn/scikit-learn/pull/25220/commits/073daf7a37dfb69c79ba4fba692be5544a2cdaba)